File: 215. Kth Largest Element in an Array/Solution.cpp
class Solution {
public:
    int findKthLargest(vector<int>& nums, int k) {
        auto cmp = [](const auto a, const auto b) { return a > b; };
        priority_queue<int, vector<int>, decltype(cmp)> pq(cmp);
        for (int n : nums) { 
            pq.push(n);
            if (pq.size() > k) pq.pop();
        }
        return pq.top();
    }
};

File: 215. Kth Largest Element in an Array/Solution.py
import heapq
class Solution:
    def findKthLargest(self, nums: List[int], k: int) -> int:
        pq = []
        for n in nums:
            heappush(pq, n)
            if len(pq) > k: heappop(pq)
        return heappop(pq)
